modules/kafka/kafka-dest-worker.c

```diff
@@ -61,32 +62,57 @@ _publish_message(KafkaDestWorker *self, LogMessage *msg)
 {
   KafkaDestDriver *owner = (KafkaDestDriver *) self->super.owner;
   int block_flag = _is_poller_thread(self) ? 0 : RD_KAFKA_MSG_F_BLOCK;
+  rd_kafka_topic_t *topic;
+  ScratchBuffersMarker marker;
+  GString *topic_name_new = scratch_buffers_alloc_and_mark(&marker);
+  gboolean is_fallback_topic = FALSE;
 
-  if (rd_kafka_produce(owner->topic,
+  if (kafka_dd_is_topic_name_a_template(owner))
+    {
+      log_template_format(owner->topic_name, msg, &owner->template_options, LTZ_SEND, self->super.seq_num, NULL,
+                          topic_name_new);
+      topic = kafka_dd_query_insert_topic(owner, topic_name_new->str);
+
+      if (!topic) //if topic name is invalid, then topic is fallback topic which is we know is defined
```

Comment:
please include the comment inside the if, we tend to use traditional `/* */` comments